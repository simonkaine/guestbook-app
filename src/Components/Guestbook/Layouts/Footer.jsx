import React from 'react';

export default function Footer() {
    return (
        <footer>
            <div>
                <p>&copy; Guestbook lab 2021</p>
            </div>
        </footer>
    )
}
