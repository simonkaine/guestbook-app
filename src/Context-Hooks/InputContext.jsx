import { useContext, createContext, useState } from "react";

const inputContext = createContext();
const InputProvider = ({children}) => {

    const [input, setInput] = useState([]);

    return (
        <inputContext.Provider value={{input, setInput}}>
            {children}
        </inputContext.Provider>
    )
}

const useInputs = () => {
    const context = useContext(inputContext)
    if(context === undefined) {
        throw new Error("Yoooo, things ain't working with your inputs dawg..")
    } return context
}

export { InputProvider, useInputs }