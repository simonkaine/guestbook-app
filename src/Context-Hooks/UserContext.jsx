import { useContext, createContext, useState } from "react";

const UserContext = createContext();
const UserProvider = ({children}) => {

    const [user, setUser] = useState('');

    return (
        <UserContext.Provider value={{user, setUser}}>
            {children}
        </UserContext.Provider>
    )
}

const useUser = () => {
    const context = useContext(UserContext)
    if(context === undefined) {
        throw new Error("Yoooo, things ain't working with your 'user' dawg..")
    } return context;
}

export { UserProvider, useUser }