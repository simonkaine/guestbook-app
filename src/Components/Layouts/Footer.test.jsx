import React from 'react'
import { render, screen } from '@testing-library/react'
import Footer from './Footer'
import { UserProvider } from '../../Context-Hooks/UserContext'

it('Should render the footer', async () => {
    const {container} = render(

        <UserProvider>
            <Footer />
        </UserProvider>

    )
    const footer = await screen.findByText(/Guestbook lab/i);
    expect(footer).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})