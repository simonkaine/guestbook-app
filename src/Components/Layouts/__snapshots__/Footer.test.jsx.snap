// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the footer 1`] = `
<div>
  <footer>
    <div>
      <p>
        Â© Guestbook lab 2021
      </p>
    </div>
  </footer>
</div>
`;
